design/user_types.go

```diff
@@ -38,6 +38,71 @@ var UpdateWorkItemPayload = a.Type("UpdateWorkItemPayload", func() {
 	a.Required("type", "fields", "version")
 })
 
+// UpdateWorkItemJSONAPIPayload defines top level structure from jsonapi specs
+// visit : http://jsonapi.org/format/#document-top-level
+var updateWorkItemJSONAPIPayload = a.Type("UpdateWorkItemJSONAPIPayload", func() {
+	a.Attribute("data", workItemDataForUpdate)
+	a.Required("data")
+})
+
+// WorkItemDataForUpdate defines how an update payload will look like
+var workItemDataForUpdate = a.Type("WorkItemDataForUpdate", func() {
+	a.Attribute("type", d.String, func() {
+		a.Enum("workitems")
+	})
+	a.Attribute("id", d.String, "ID of the work item which is being updated", func() {
+		a.Example("42")
+	})
+	a.Attribute("attributes", a.HashOf(d.String, d.Any), func() {
+		a.Example(map[string]interface{}{"version": "1", "system.state": "new", "system.title": "Example story"})
+	})
+	a.Attribute("relationships", workItemRelationships)
+	// relationships must be required becasue we MUST have workItemType during PATCh
+	a.Required("type", "id", "attributes")
```

Comment:
I'm not sure that `attributes` has to be a required field. If you just want to update the relationship (e.g. the assignee), then you only need to touch the `relationships` element? What do you think?